lib/util/lifted/influx/httpd/handler_metrics.go

```diff
@@ -0,0 +1,263 @@
+					zap.String("query", qDuration.Query))
+			}
+		}()
+	}
+
+	// Check authorization.
+	err = h.checkAuthorization(user, q, db)
+	if err != nil {
+		return nil, fmt.Errorf("error authorizing query: " + err.Error())
+	}
+	// Parse chunk size. Use default if not provided or unparsable.
+	chunked, chunkSize, innerChunkSize, err := h.parseChunkSize(r)
+	if err != nil {
+		return nil, err
+	}
+	// Parse whether this is an async command.
+	async := r.FormValue("async") == "true"
+
+	opts := query.ExecutionOptions{
+		Database:        db,
+		RetentionPolicy: r.FormValue("rp"),
+		ChunkSize:       chunkSize,
+		Chunked:         chunked,
+		ReadOnly:        r.Method == "GET",
+		NodeID:          nodeID,
+		InnerChunkSize:  innerChunkSize,
+		ParallelQuery:   atomic.LoadInt32(&syscontrol.ParallelQueryInBatch) == 1,
+		Quiet:           true,
+		Authorizer:      h.getAuthorizer(user),
+	}
+
+	// Make sure if the client disconnects we signal the query to abort
+	var closing chan struct{}
+	if !async {
+		closing = make(chan struct{})
+		done := make(chan struct{})
+
+		opts.AbortCh = closing
+		defer func() {
+			close(done)
+		}()
+		go func() {
+			select {
+			case <-done:
+			case <-r.Context().Done():
+			}
+			close(closing)
+		}()
+	}
+
+	// Execute query
+	results := h.QueryExecutor.ExecuteQuery(q, opts, closing, qDuration)
+
+	//// if we're not chunking, this will be the in memory buffer for all results before sending to client
+	stmtID2Result := make(map[int]*query.Result)
+	//
+	//// Status header is OK once this point is reached.
+	//// Attempt to flush the header immediately so the client gets the header information
+	//// and knows the query was accepted.
+	//// pull all results from the channel
+	rows := 0
+	for r := range results {
+		// Ignore nil results.
+		if r == nil {
+			continue
+		}
+
+		rows = h.getResultRowsCnt(r, rows)
+		if !h.updateStmtId2Result(r, stmtID2Result) {
+			continue
+		}
+		//
+		//	// Drop out of this loop and do not process further results when we hit the row limit.
+		if h.Config.MaxRowLimit > 0 && rows >= h.Config.MaxRowLimit {
+			// If the result is marked as partial, remove that partial marking
+			// here. While the series is partial and we would normally have
+			// tried to return the rest in the next chunk, we are not using
+			// chunking and are truncating the series so we don't want to
+			// signal to the client that we plan on sending another JSON blob
+			// with another result.  The series, on the other hand, still
+			// returns partial true if it was truncated or had more data to
+			// send in a future chunk.
+			r.Partial = false
+			break
+		}
+	}
+
+	resp := h.getStmtResult(stmtID2Result)
+	if resp.Results[0].Err != nil {
+		return nil, resp.Results[0].Err
+	}
+	metricsMap := make(map[string]interface{})
+
+	for i, key := range resp.Results[0].Series[0].Columns {
+		metricsMap[key] = resp.Results[0].Series[0].Values[0][i]
+	}
+	labelValues := make(map[string]string)
```

Comment:
optimized: allocate the cap of the map in advance, when the Tags length is known,

Reply:
OK, this code logic has been modified.